src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -154,19 +190,36 @@ public class ClassDeclaration : ProprietableBlockDeclaration
         }
     }
 
-    public CodeProperty? GetOriginalPropertyDefinedFromBaseType(string propertyName)
+    public CodeProperty? GetOriginalPropertyDefinedFromBaseType(string serializationName)
     {
-        ArgumentException.ThrowIfNullOrEmpty(propertyName);
+        ArgumentException.ThrowIfNullOrEmpty(serializationName);
 
         if (inherits is CodeType currentInheritsType &&
             currentInheritsType.TypeDefinition is CodeClass currentParentClass)
-            if (currentParentClass.FindChildByName<CodeProperty>(propertyName) is CodeProperty currentProperty && !currentProperty.ExistsInBaseType)
+            if (currentParentClass.FindChild<CodeProperty>(x => x.WireName == serializationName) is CodeProperty currentProperty && !currentProperty.ExistsInBaseType)
                 return currentProperty;
             else
-                return currentParentClass.StartBlock.GetOriginalPropertyDefinedFromBaseType(propertyName);
+                return currentParentClass.StartBlock.GetOriginalPropertyDefinedFromBaseType(serializationName);
         return default;
     }
+    public CodeProperty? FindPropertyByNameInTypeHierarchy(string propertyName)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(propertyName);
 
+        if (Parent is CodeClass thisClass)
```

Comment:
moving to the code class instead of starting block would allow you to remove this condition

Reply:
Yes, you are right. I've moved the method now.